src/interrupts/gyro_int.rs

```diff
@@ -0,0 +1,544 @@
+            xaxis_low_event: match reg_data & G_INT_Bitmasks::XL_G {
+                0 => false,
+                _ => true,
+            },
+            /// Roll (Y-axis) high event has occurred
+            yaxis_high_event: match reg_data & G_INT_Bitmasks::YH_G {
+                0 => false,
+                _ => true,
+            },
+            /// Roll (Y-axis) low event has occurred
+            yaxis_low_event: match reg_data & G_INT_Bitmasks::YL_G {
+                0 => false,
+                _ => true,
+            },
+            /// Yaw (Z-axis) high event has occurred
+            zaxis_high_event: match reg_data & G_INT_Bitmasks::ZH_G {
+                0 => false,
+                _ => true,
+            },
+            /// Yaw (Z-axis) low event has occurred
+            zaxis_low_event: match reg_data & G_INT_Bitmasks::ZL_G {
+                0 => false,
+                _ => true,
+            },                
+        };
+        Ok(status)
+    }
+
+    /// Set gyroscope reference value for digital high-pass filter.
+    pub fn set_hipass_ref(&mut self, value: u8) -> Result<(), T::Error> {
+        self.interface.write(Sensor::Gyro, register::AG::REFERENCE_G.addr(), value)?;
+        Ok(())
+    }
+
+    /// Get gyroscope reference value for digital high-pass filter.
+    pub fn read_hipass_ref(&mut self) -> Result<u8, T::Error> {
+        let data: u8 = self.read_register(Sensor::Gyro, register::AG::REFERENCE_G.addr())?;
+        Ok(data)
+    }
+    
+    /// Gyroscope interrupt duration
+    /// Enable/disable wait function and define duration (for how many samples to wait before exiting interrupt)    
+    pub fn gyro_int_duration(&mut self, wait: FLAG, duration: u8) -> Result<(), T::Error> {
+        // read the current value of the register
+        
+        let mut reg_value = self.read_register(Sensor::Gyro, register::AG::INT_GEN_DUR_G.addr())?;
+
+        match wait {
+            FLAG::Enabled => reg_value & !0b1000_0000 | 0b1000_0000, // set bit
+            FLAG::Disabled => reg_value & !0b1000_0000, // clear bit
+        };
+
+        let duration: u8 = match duration { // clamp duration to 7 bit values
+            0..=127 => duration,
+            _ => 127,
+        };
+
+        reg_value &= !0b0111_1111; // clear the lowest 7 bits
+
+        reg_value |= duration; 
+
+
+        self.interface.write(Sensor::Gyro, register::AG::INT_GEN_DUR_G.addr(), reg_value)?;
+
+        Ok(())
+
+    }
+
+    /// Set gyroscope mode during inactivity, activation threshold and inactivity duration
+    pub fn set_gyro_inactivity(&mut self, setting: FLAG, threshold: u8, duration: u8) -> Result<(), T::Error> {
+        let mut data: u8 = 0;
+
+        match setting {
+            FLAG::Enabled => data | (1 << 7),
+            FLAG::Disabled => data,
+        };
+
+        match threshold {
+            0..=127 => data | threshold,
+            _ => data | 127,
+        };
```

Comment:
Are you trying to modify `data`? I don't think it is modified at all. If you drop `mut`, it should still compile.

Did you mean something like this?

```rust
        data = match setting {
            FLAG::Enabled => data | (1 << 7),
            FLAG::Disabled => data,
        };
```

Reply:
I'm not sure if I understand. The idea is to start with `data = 0`, that is: all the bits are cleared, and then according to the values of `setting` and `threshold` some of those bits would either be set or not. 
So if `setting` is `Disabled` and the `threshold` is set to 0, then it will stay 0, otherwise something will be modified. I suppose that maybe your example was meant to be:

```
 data = match setting {
            FLAG::Enabled => (1 << 7),
            FLAG::Disabled => 0,
        };
```
?

Reply:
I understand what you want to do. But your original code does not do that. `data` is always `0` because you don't assign a new value.

Reply:
OK, now I see it... OK, will fix that, too, and need to check if there isn't more of that in the code :facepalm:

Reply:
OK, fixed this one.